tests/integration/advanced/test_custom_modules.py

```diff
@@ -62,3 +62,45 @@ def test_0_add_throwing_cleaner(self):
 
         raise Exception('Predictor did not contain modified function!')
 
+    def test_1_add_analyzer_block(self):
+
+        mname = 'custom_analyzers'
+        cname = 'ExampleAnalysis'
+        module_code = f"""
+from lightwood.analysis.base import BaseAnalysisBlock
+
+class {cname}(BaseAnalysisBlock):
+    def __init__(self):
+        super().__init__(deps=None)
+
+    def analyze(self, info, **kwargs):
+        info['test'] = 'test'
+        return info
+
+    def explain(self, row_insights, global_insights, **kwargs):
+        row_insights['test'] = 'test'
+        return row_insights, global_insights
+"""
+        create_custom_module(os.path.join(mdir, f'{mname}.py'), module_code)
+
+        # Create base json ai
+        df = FileDS('tests/data/hdi.csv').df.iloc[0:400]
+        json_ai = json_ai_from_problem(df, ProblemDefinition.from_dict({'target': 'Development Index', 'time_aim': 20}))
+
+        # modify it
+        json_ai_dump = json_ai.to_dict()
+        json_ai_dump['analysis_blocks'] = [{
+            'module': f'{mname}.{cname}',
+            'args': {}
+        }]
+
+        json_ai = JsonAI.from_dict(json_ai_dump)
+
+        # create a predictor from it
+        code = code_from_json_ai(json_ai)
+        predictor = predictor_from_code(code)
+        predictor.learn(df)
+        row_insights = predictor.predict(df)
```

Comment:
I thought this test had a `.save` and a `.load` called on the predictor object (and hence that's why you spotted the import bug)... anyway, outside the scope of this test or this PR to do this but made this issue: https://github.com/mindsdb/lightwood/issues/556 | so thanks for pointing this out :) \

Reply:
Yeah, caught it with an external test script that did `.save()` and `.load()`. I agree on following up with a PR for #556.